Test serve start config

gsnedders opened the pull request:
Mostly throwing this up because I think this is a terrible test but I don't know how to do better? And it shows a real issue.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/w3c/web-platform-tests/9726)
<!-- Reviewable:end -->

gsnedders committed:
Add test to check that serve.start gets called with right host

w3c-bots commented:
# Build BROKEN

Started: 2018-03-01 02:31:22
Finished: 2018-03-01 02:43:19


<h2>Failing Jobs</h2>
<ul>

<li>tools_unittest in py27</li>

<li>tools_unittest in pypy</li>

</ul>




View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/25067)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/347578438)

jgraham commented on tools/wptrunner/wptrunner/tests/test_products.py:
```diff
@@ -3,12 +3,23 @@
 import pytest
 
-sys.path.insert(0, join(dirname(__file__), "..", ".."))
+sys.path.insert(0, join(dirname(__file__), "..", "..", "..", "..")) # /
```
Why do you need both of these on the path?

gsnedders replied:
Um, because I didn't change everything to use `from tools`, seemingly. Fixed.

jgraham reviewed (commented):
I don't love this either, but I guess it's OK.

gsnedders committed:
Test all wptrunner environments

gsnedders committed:
Rewrite the code to parametrise over products

gsnedders committed:
Fix #9750: bind_hostname needs to be a Python bool

db55c167d484d0e34fd618960887161522280d47 fixed this for Servo;
"true"/"false" worked when it was interpolated as JSON, but that is no
longer the case since cb2d75f (#9480).

gsnedders commented:
Note that this covers both #9750 and #9651.

gsnedders commented:
@jgraham if it's okay can we land it?

jgraham commented on tools/wptrunner/wptrunner/browsers/chrome_android.py:
```diff
@@ -68,7 +68,7 @@ def env_extras(**kwargs):
 
 def env_options():
-    return {"bind_hostname": "true"}
+    return {"bind_hostname": True}
```
Let's just make `True` the default and remove all of these

gsnedders replied:
done

gsnedders committed:
Use default value of bind_hostname unless we need to override it
